stone_age/game_board/reward_menu.py

```diff
@@ -0,0 +1,37 @@
+import json
+from typing import List, Any, Mapping
+from stone_age.simple_types import Effect, HasAction, PlayerOrder
+from stone_age.game_board.simple_types import Player
+from stone_age.interfaces import InterfaceTakeReward
+
+
+class RewardMenu(InterfaceTakeReward):
+    def __init__(self,
+                 players: Mapping[PlayerOrder, Player]) -> None:
+        self.menu: List[Effect] = []
+        self.players = players
+
+    def initiate(self, menu: List[Effect]) -> None:
+        self.menu = menu
+
+    def take_reward(self, player: PlayerOrder, reward: Effect) -> bool:
+        if reward in self.menu:
+            self.players[player].player_board.give_effect([reward])
+            self.menu.remove(reward)
```

Comment:
When taking a reward, I think there should be a check whether the player has already taken a reward. It can be done by adding one more list as attribute, where you keep players, who haven't taken reward yet (don't forget to reset it in `initiate` method, if you do it this way).